wordpress.org/public_html/wp-content/plugins/theme-directory/class-wporg-themes-upload.php

```diff
@@ -341,10 +342,10 @@ public function process_upload( $file_upload ) {
 
 			if ( ! $is_allowed_to_upload_for_theme ) {
 				/* translators: 1: theme slug, 2: style.css */
-				return sprintf( __( 'There is already a theme called %1$s by a different author. Please change the name of your theme in %2$s and upload it again.', 'wporg-themes' ),
+				array_push( $style_errors, sprintf( __( 'There is already a theme called %1$s by a different author. Please change the name of your theme in %2$s and upload it again.', 'wporg-themes' ),
```

Comment:
This error is pretty much a show-stopping "Any other error doesn't matter" case.. but I think the UX of getting given a full error list is worthwhile over burrying a critical block like this half way through the list.

Reply:
Are you suggesting I split it out? Or it's okay as is?

I considered moving this up to the top under the check for a slug but I just deferred to its current place out of not quite knowing why it's where it is :)

Reply:
I suggest splitting it out if it is not too much trouble.
But still show any other error that comes after it.

Reply:
Let's see what feedback we get after this change to see what effect it has. It might make it more confusing for some authors when they get both "you can't do this" and "fix this before you upload" in the same error message, but for existing themes, I think that's likely to be less common.